### PR TITLE
[PackageBuilder] Fix crash on linux when there is only one test module and is C

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
PackageBuilderTester found another bug 💃

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
depends on #538 and #531

File: Tests/PackageLoading/ConventionTests.swift [hunk 1]
```diff
@@ -96,13 +96,38 @@ class ConventionTests: XCTestCase {
...
+                                "/MyPackage/Tests/MyPackage/abc.c")
+
+        PackageBuilderTester("/MyPackage", in: fs) { result in
+            result.checkModule("MyPackage") { moduleResult in
+                moduleResult.check(type: .executable, isTest: false)
+                moduleResult.checkSources("main.swift")
+                moduleResult.checkSources(root: "/MyPackage")
+            }
+
+            result.checkModule("MyPackageTestSuite") { moduleResult in
```
Comment (ddunbar) [inline Tests/PackageLoading/ConventionTests.swift]: Doesn't this test fail on Linux, for exactly the reasons this test is trying to test?

Comment (aciidb0mb3r) [inline Tests/PackageLoading/ConventionTests.swift]: The module is created, it is ignored and a warning is emitted while making the products

Comment (ddunbar) [inline Tests/PackageLoading/ConventionTests.swift]: Ah, got it.

**Pull request merged**